openedx/core/djangoapps/content_libraries/api.py

```diff
@@ -835,9 +835,11 @@ def validate_can_add_block_to_library(
     return content_library, usage_key
 
 
-def create_library_block(library_key, block_type, definition_id, user_id=None):
+def create_library_block(library_key, block_type, definition_id, user_id=None, collection_key=None):
     """
     Create a new XBlock in this library of the specified type (e.g. "html").
+
+    Optionally associates the newly created xblock with given collection_key.
```

Comment:
Q: Is it important to assign a collection and create a block in a single API call, as opposed to two API calls? We don't have an API to set any other things during creation, like tags, display name, content, etc. - those are done by follow-up calls.

Reply:
That's a good point -- there's nothing special about collections or how they're indexed that requires doing this in a single step.

We'd still need the filterable fields update, but not the rest of this PR.. @navinkarkera what do you think?

Reply:
@bradenmacdonald, @pomegranited Yes, that makes sense. I have removed the commit which added collections during creation and updated frontend PR to make an additional api call.

Reply:
Thanks! I definitely feel better keeping it simple for now.